src/main/java/com/faforever/client/map/management/MapsManagementController.java

```diff
@@ -0,0 +1,180 @@
+import javafx.beans.property.SimpleListProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.ChoiceBox;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.image.Image;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Region;
+import javafx.util.StringConverter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.artifact.versioning.ComparableVersion;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+@Slf4j
+public class MapsManagementController implements Controller<Node> {
+
+  public GridPane root;
+  public ChoiceBox<MapFilter> filterMapsChoiceBox;
+  public Button removeButton;
+  public Button closeButton;
+  public TableView<MapBean> tableView;
+  public TableColumn<MapBean, Image> previewMapColumn;
+  public TableColumn<MapBean, MapBean> mapNameColumn;
+  public TableColumn<MapBean, MapBean> checkboxColumn;
+
+  private final MapService mapService;
+  private final I18n i18n;
+  private final NotificationService notificationService;
+
+  private final ObservableList<MapBean> selectedMaps = FXCollections.observableArrayList();
+
+  private FilteredList<MapBean> maps = new FilteredList<>(FXCollections.emptyObservableList());
+  private List<String> duplicatedMapNames; // duplicated map names but with different versions
+  private Runnable closeButtonAction;
+
+  @Override
+  public void initialize() {
+    settingChoiceBox();
+    settingTableView();
+    JavaFxUtil.bind(removeButton.disableProperty(), new SimpleListProperty<>(selectedMaps).emptyProperty());
+    updateTableContent();
+  }
+
+  private void updateTableContent() {
+    List<MapBean> clonedList = new ArrayList<>(mapService.getInstalledMaps());
+    List<String> mapNames = clonedList.stream().map(MapBean::getDisplayName).collect(Collectors.toList());
+    this.maps = new FilteredList<>(FXCollections.observableArrayList(clonedList).sorted(), getPredicateByCurrentFilter());
+    this.duplicatedMapNames = clonedList.stream()
+        .filter(map -> map.getVersion() != null && Collections.frequency(mapNames, map.getDisplayName()) > 1)
+        .map(MapBean::getDisplayName).distinct().collect(Collectors.toList());
+    tableView.setItems(this.maps);
```

Comment:
You don't need create a clonedList here since you are using it to create streams. Also if you use the InstalledMaps Observable List itself the filteredList should update if the map list changes